src/webui/api/synccontroller.cpp

```diff
@@ -119,9 +117,9 @@ namespace
     const QString KEY_FULL_UPDATE = u"full_update"_s;
     const QString KEY_RESPONSE_ID = u"rid"_s;
 
-    void processMap(const QVariantMap &prevData, const QVariantMap &data, QVariantMap &syncData);
-    void processHash(QVariantHash prevData, const QVariantHash &data, QVariantMap &syncData, QVariantList &removedItems);
-    void processList(QVariantList prevData, const QVariantList &data, QVariantList &syncData, QVariantList &removedItems);
+    QVariantMap processMap(const QVariantMap &prevData, const QVariantMap &data);
+    std::pair<QVariantMap, QVariantList> processHash(QVariantHash prevData, const QVariantHash &data);
+    std::pair<QVariantList, QVariantList> processList(QVariantList prevData, const QVariantList &data);
```

Comment:
Will it cause performance regression? The last time I looked into this, any change will cause a non-trivial performance slow down.
Could you do some benchmark to ensure it won't happen?

Reply:
> Will it cause performance regression?

What exactly should cause performance regression? Using a return value instead of an output parameter could do this?

Reply:
On a closer look, it seems it just changed from local variable to return values, right? If so, then it should be fine.

Reply:
> On a closer look, it seems it just changed from local variable to return values, right?

Yes. And added a couple of checks to see if the returned value is empty.